clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java

```diff
@@ -503,13 +504,16 @@ public int read(ByteBuffer dst) throws IOException {
             read = readFromAppBuffer(dst);
         }
 
+        boolean readFromNetwork = false;
         boolean isClosed = false;
         // Each loop reads at most once from the socket.
         while (dst.remaining() > 0) {
             int netread = 0;
             netReadBuffer = Utils.ensureCapacity(netReadBuffer, netReadBufferSize());
             if (netReadBuffer.remaining() > 0)
                 netread = readFromSocketChannel();
+            if (netread > 0)
```

Comment:
performance nit - this if can be moved inside the previous if, for some infinitesimal performance gain?